receiver/otlpreceiver/otlp_test.go

```diff
@@ -542,6 +542,31 @@ func TestHTTPNewPortAlreadyUsed(t *testing.T) {
 	require.Error(t, r.Start(context.Background(), componenttest.NewNopHost()))
 }
 
+func TestReceiverLifecycle(t *testing.T) {
+	endpointGrpc := testutil.GetAvailableLocalAddress(t)
+	endpointHTTP := testutil.GetAvailableLocalAddress(t)
+
+	tracesSink := new(consumertest.TracesSink)
+	metricsSink := new(consumertest.MetricsSink)
+
+	// Create OTLP receiver with gRPC and HTTP protocols.
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig().(*Config)
+	cfg.GRPC.NetAddr.Endpoint = endpointGrpc
+	cfg.HTTP.Endpoint = endpointHTTP
+
+	fstReceiver := newReceiver(t, factory, cfg, tracesSink, metricsSink)
+	mh := newAssertNoErrorHost(t)
```

Comment:
This test is flaky because `ReportFatalError` is called async, how can you ensure that it does not happen? You have to ensure that there is no reference to the host, or there is no other way that this can happen.

Reply:
Adding a `sync.WaitGroup`  to ensure that Shutdown only returns after the server goroutines finish. Notice that the code doesn't call `ReportFatalError` anymore for `grpc.ErrServerStopped` and `http.ErrServerStopped` which are expected on shutdown.

I will be removing this local test in favor of the one at `defaultcomponents`.